tests/baselines/reference/tsbuildWatch/moduleResolution/resolves-specifier-in-output-declaration-file-from-referenced-project-correctly-with-cts-and-mts-extensions.js

```diff
@@ -315,12 +315,67 @@ File '/user/username/projects/myproject/packages/pkg2/build/const.d.cts' exist -
+
+[7m1[0m import type { TheNum } from 'pkg2'
+[7m [0m [91m                            ~~~~~~[0m
+
+  [96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m
+    [7m1[0m import type { TheNum } from 'pkg2'
+    [7m [0m [96m                            ~~~~~~[0m
+    To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
+
+[96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m - [91merror[0m[90m TS1479: [0mThe current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("pkg2")' call instead.
+
+[7m1[0m import type { TheNum } from 'pkg2'
+[7m [0m [91m                            ~~~~~~[0m
+
+  [96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m
+    [7m1[0m import type { TheNum } from 'pkg2'
+    [7m [0m [96m                            ~~~~~~[0m
+    To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
+
+[96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m - [91merror[0m[90m TS1479: [0mThe current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("pkg2")' call instead.
+
+[7m1[0m import type { TheNum } from 'pkg2'
+[7m [0m [91m                            ~~~~~~[0m
+
+  [96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m
+    [7m1[0m import type { TheNum } from 'pkg2'
+    [7m [0m [96m                            ~~~~~~[0m
+    To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
+
+[96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m - [91merror[0m[90m TS1479: [0mThe current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("pkg2")' call instead.
+
+[7m1[0m import type { TheNum } from 'pkg2'
+[7m [0m [91m                            ~~~~~~[0m
+
+  [96mpackages/pkg1/index.ts[0m:[93m1[0m:[93m29[0m
+    [7m1[0m import type { TheNum } from 'pkg2'
+    [7m [0m [96m                            ~~~~~~[0m
+    To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
+
+[[90m12:01:16 AM[0m] Found 6 errors. Watching for file changes.
```

Comment:
@sheetalkamat this error message seem to be duplicated in tsbuildWatch and tsserver unit tests and I have no idea why—any ideas?

Reply:
To me it seems like for some reason they are not getting de-dupped?

Reply:
It seems like errors with related info don’t get deduped because the deduping logic tends to happen before attaching the related info, so the diagnostic being added doesn’t appear to be equal to the ones already in the collection yet. I switched to a message chain since related info is better suited for pointing to a different location, but we should maybe look and see if this is happening elsewhere.